docs/src/pages/components/data-grid/editing/editing.md

```diff
@@ -5,7 +5,129 @@ components: DataGrid, XGrid
+
+{{"demo": "pages/components/data-grid/editing/ValueGetterGrid.js", "bg": "inline"}}
+
+### Client-side validation
+
+Follow the following steps to validate the value in the cells.
+
+- Set the event handler `onEditCellChange`. It's invoked when a change is triggered by the edit input component.
+- Set the event handler `onEditCellChangeCommitted` to validate or persist the new value. This handler is invoked when an end-user requests a change to be committed and before the cell reverts to view mode.
+
+Alternatively, you can use the `GridEditRowsModel` state mentioned in the [Control editing](#control-editing) section.
+
+{{"demo": "pages/components/data-grid/editing/ValidateRowModelControlGrid.js", "bg": "inline"}}
+
+#### Using apiRef [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-pro/)
+
+{{"demo": "pages/components/data-grid/editing/ValidateCellApiRefGrid.js", "bg": "inline"}}
+
+### Server-side validation
+
+Server-side validation works like client-side [validation](#validation).
+
+- Use the `GridEditRowsModel` mentioned in the [Control editing](#control-editing) section.
+- Or set the event handler `onEditCellChange` for `keydown` validation
+- Set the event handler `onEditCellChangeCommitted` to validate and update the server when the change is committed.
+
+**Note:** To prevent the default client-side behavior, use `event.stopPropagation()`.
+
+This demo shows how you can validate a username asynchronously and prevent the user from committing the value while validating.
+
+{{"demo": "pages/components/data-grid/editing/ValidateServerNameGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom edit component
+
+To customize the edit component of a column, set the `renderEditCell` function available in the column definition `GridColDef`.
+
+The demo lets you edit the ratings by double-clicking the cell.
+
+{{"demo": "pages/components/data-grid/editing/RenderRatingEditCellGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Edit using external button
+
+You can override the default [start editing](#start-editing) triggers using the `event.stopPropagation()` API on the synthetic React events.
+
+{{"demo": "pages/components/data-grid/editing/StartEditButtonGrid.js", "bg": "inline"}}
+
+### Events
+
+The editing feature leverages the event capability of the grid and the apiRef.
+The following events can be imported and used to customize the edition:
+
+- `GRID_CELL_EDIT_ENTER`: emitted when the cell turns to edit mode.
+- `GRID_CELL_EDIT_EXIT`: emitted when the cell turns back to view mode.
+- `GRID_CELL_EDIT_PROPS_CHANGE`: emitted when the edit input change.
+- `GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED`: emitted when the edit input change is submitted.
+
+Catching events can be used to add a callback after an event while ignoring its triggers.
+
+_The demo shows how to catch the start & end edit events to log which cell is editing in an info message._
```

Comment:
More of a general writing docs convention, We never used the _italic_ to describe the demos so far (AFAIK), so it seems more consistent with:

```suggestion
The demo shows how to catch the start & end edit events to log which cell is editing in an info message:
```